discovery/sync_manager.go

```diff
@@ -108,6 +103,18 @@ type SyncManager struct {
 	// _once_ with a peer during the SyncManager's startup.
 	historicalSync sync.Once
 
+	// newSyncers is a channel we'll use to process requests to create
+	// GossipSyncers for newly connected peers.
+	newSyncers chan *newSyncer
+
+	// staleSyncers is a channel we'll use to process requests to tear down
+	// GossipSyncers for disconnected peers.
+	staleSyncers chan *staleSyncer
+
+	// syncersMu guards the read and write access to the activeSyncers and
+	// inactiveSyncers maps below.
+	syncersMu sync.Mutex
```

Comment:
Correct. I went with the mutex to avoid the additional internal messages within the event loop, but I'm not opposed to adding them in order to remove it.

Reply:
i'm fine as is